plugin/core/windows.py

```diff
@@ -430,16 +430,18 @@ def _start_client(self, config: ClientConfig, file_path: str) -> None:
             debug('Already starting on this window:', config.name)
             return
 
-        if not self._handlers.on_start(config.name, self._window):
+        workspace_paths = sorted_workspace_paths(self._workspace.folders, file_path)
+        startable_folders = self._handlers.on_start(config.name, self._window, workspace_paths, file_path)
+
+        if not startable_folders:
```

Comment:
It doesn't quite feel right to return a list of folders from can_start.

What would the meaning be when the server does and doesn't support workspaceFolders?
 - if it doesn't then we start one server per window, even with multiple folders, right?
With this here, we would start it if at least one folder has flow config.
Would that be a problem for the other folder(s) since those wouldn't have a config?
 - If it does support workspaces then I guess can_start will receive a list with just one folder right? If yes then that case should be fine.

But then it appears that the returned list of folders is not used anyway so why not just return True/False?

Reply:
Or are we gonna get multiple folders when the server supports workspaceFolders? If so then we would probably have to answer yes/no per-folder.

Reply:
We don't know if the server supports workspace folders at this point (before it has replied to `initialize`), so here the handler will always receive all the folders in the project.
The possible responses with this use are:
* All `workspace_paths` returned (server doesn't care what folders are opened)
* Some `workspace_paths` returned (server is picky about what folders it supports, eg flow lsp)
* No `workspace_paths` are returned (server can't start at all - eg. is not installed)

Reply:
> What would the meaning be when the server does and doesn't support workspaceFolders?

> Or are we gonna get multiple folders when the server supports workspaceFolders? If so then we would probably have to answer yes/no per-folder.

In an LSP-foo package, you know exactly which server you're dealing with, so you know whether that server supports workspace folders or not.

Reply:
And what about that part?

> But then it appears that the returned list of folders is not used anyway so why not just return True/False?

What happens with Julia server if one of the folders doesn't have configuration file. Will it crash anyway?
Does its behavior depend on which file is opened first (from which folder)?